server/src/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -4600,14 +4602,18 @@ public UserVm createVirtualMachine(DeployVMCmd cmd) throws InsufficientCapacityE
 
         DataCenter zone = _entityMgr.findById(DataCenter.class, zoneId);
         if (zone == null) {
-            throw new InvalidParameterValueException("Unable to find zone by id=" + zoneId);
+            CloudRuntimeException ex =  new InvalidParameterValueException("Unable to find zone by id=" + zoneId);
```

Comment:
Does it make sense to instead, get/set the context/jobid in the exception constructor if CallContext is available to it?

Reply:
thanks, yes that's why I created the CloudRuntimeException(String callId, String message) but i haven't created a InvalidParameterValueException(String callId, String message) yet.